Libraries/libbom/Headers/bom/bom.h

```diff
@@ -39,7 +39,7 @@ bom_context_memory_file(const char *fn, bool writeable, size_t minimum_size);
 struct bom_context;
 
 struct bom_context *
-bom_alloc_empty(struct bom_context_memory memory);
+bom_alloc_size(struct bom_context_memory memory, uint32_t index_count);
```

Comment:
Maybe still call these `bom_alloc_empty` and `bom_tree_alloc_empty`, since they create empty BOMs? I think "size" doesn't communicate as well that it's creating a *new* BOM rather than loading one in.